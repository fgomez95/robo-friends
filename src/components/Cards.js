import React from 'react';
import Card from './Card';

const Cards = ({ data }) => {
    let cards;
    console.log(data);
    if(data.length){
        cards = data.map(robot => {
            return (
                <Card
                    name={robot.name}
                    email={robot.email}
                    image={`https://robohash.org/${robot.id}?set=set2?size=200x200`}
                    key={robot.id}
                />
            );
        });
    }
    else { 
        cards = <p>No items yet</p>;
    }
    return (
        <div>
            {cards}
        </div>
    );
};

export default Cards;