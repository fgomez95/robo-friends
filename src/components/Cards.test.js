import {shallow} from 'enzyme';
import Cards from './Cards';
import React from 'react';

const robots = [
  {
    id: 1,
    name: 'john snow',
    email: 'jsnow@gmail',
  }
]
it('render without crashing', () => {
  expect(shallow(<Cards data={robots}/>)).toMatchSnapshot();
});