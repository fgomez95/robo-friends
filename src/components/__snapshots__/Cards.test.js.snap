// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards
    data={
      Array [
        Object {
          "email": "jsnow@gmail",
          "id": 1,
          "name": "john snow",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <card
          email="jsnow@gmail"
          image="https://robohash.org/1?set=set2?size=200x200"
          name="john snow"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "jsnow@gmail",
          "image": "https://robohash.org/1?set=set2?size=200x200",
          "name": "john snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <card
            email="jsnow@gmail"
            image="https://robohash.org/1?set=set2?size=200x200"
            name="john snow"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "jsnow@gmail",
            "image": "https://robohash.org/1?set=set2?size=200x200",
            "name": "john snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
